pkg/daemon/ceph/client/pool.go

```diff
@@ -333,29 +333,49 @@ func setCommonPoolProperties(context *clusterd.Context, clusterInfo *ClusterInfo
 			}
 		}
 	} else {
-		if pool.Mirroring.Mode == "pool" {
-			// Remove storage cluster peers
-			mirrorInfo, err := GetPoolMirroringInfo(context, clusterInfo, pool.Name)
+		// Remove storage cluster peers
+		mirrorInfo, err := GetPoolMirroringInfo(context, clusterInfo, pool.Name)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
+		}
+		for _, peer := range mirrorInfo.Peers {
+			if peer.UUID != "" {
+				err := removeClusterPeer(context, clusterInfo, pool.Name, peer.UUID)
+				if err != nil {
+					return errors.Wrapf(err, "failed to remove cluster peer with UUID %q for the pool %q", peer.UUID, pool.Name)
+				}
+			}
+		}
+
+		if mirrorInfo.Mode == "pool" {
+			// Disable mirroring on pool
+			err = disablePoolMirroring(context, clusterInfo, pool.Name)
+			if err != nil {
+				return errors.Wrapf(err, "failed to disable mirroring for pool %q", pool.Name)
+			}
+		} else if mirrorInfo.Mode == "image" {
+			// Disable mirroring on pool images
+			images, err := gePoolImages(context, clusterInfo, pool.Name)
 			if err != nil {
-				return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
+				return errors.Wrapf(err, "failed to get images for pool %q", pool.Name)
 			}
-			for _, peer := range mirrorInfo.Peers {
-				if peer.UUID != "" {
-					err := removeClusterPeer(context, clusterInfo, pool.Name, peer.UUID)
-					if err != nil {
-						return errors.Wrapf(err, "failed to remove cluster peer with UUID %q for the pool %q", peer.UUID, pool.Name)
-					}
+
+			for _, image := range images {
```

Comment:
There could be a lot of images. I wonder if we need to clean this up in a job, similar to how it's proposed for force deletion of the subvolume groups.

Reply:
made changes to disable mirroring on only those images were it was enabled.

Reply:
Since we still don't have the force deletion of the subvolume PR yet, I propose that disabling mirroring for images in a job (if required), can be taken up once the other PR is ready.